TheSpaceRangers/APIAuthentication
# Dev admin

# Merge dev-admin into dev

## Description

This pull request aims to merge the dev-admin branch into the dev branch. Below are the key features and changes introduced in this branch:

# Key Features:

1. Admin User Service and Controller:

* Implemented functionalities for administrators to manage users.
* Added endpoints for creating, updating, and deleting users.

2. Admin Role Service and Controller:

* Added features for administrators to manage roles.
* Implemented endpoints for creating, updating, and deleting roles.

3. Admin User Roles Service and Controller:

* Developed services and controllers for managing user roles by the administrator.
* Added functionalities to assign and remove roles for multiple users.

4. Admin Login History Service and Controller:

* Implemented functionalities for administrators to manage and view user login histories.
* Added endpoints for retrieving login history details.

5. Reset Service and Controller:

* Created functionalities for sending password reset emails to users.
* Implemented endpoints for users to reset their passwords via email.

## How to Test:

1. Unit Tests:

* Run mvn test to execute all unit tests.
* Ensure that all tests pass successfully.

2. Integration Tests:

* Run the application and use Postman or similar tools to test the new endpoints.
* Validate the role and user management functionalities by adding, updating, and removing users and roles.
